lib/pi_piper/sysfs/sysfs.rb

```diff
@@ -0,0 +1,73 @@
+module PiPiper
+  class Sysfs < Driver
+
+    def initialize
+      @exported_pins = Set.new
+    end
+
+    def close
+      unexport_all
+      @exported_pins.empty? 
+    end
+    
+# Support GPIO pins
+    GPIO_HIGH = 1
+    GPIO_LOW  = 0
+    
+    def pin_direction(pin, direction)
+      export(pin)
+      raise ArgumentError, "Pin #{pin} not exported" unless exported?(pin)
+      raise ArgumentError, "direction should be :in or :out" unless [:in, :out].include? direction
+
+      File.write("/sys/class/gpio/gpio#{pin}/direction", direction)
+    end
+
+    def pin_read(pin)
+      raise ArgumentError, "Pin #{pin} not exported" unless exported?(pin)
+      File.read("/sys/class/gpio/gpio#{pin}/value").to_i
+    end
+
+    def pin_write(pin, value)
+      raise ArgumentError, "value should be GPIO_HIGH or GPIO_LOW" unless [GPIO_LOW, GPIO_HIGH].include? value
+      raise ArgumentError, "Pin #{pin} not exported" unless exported?(pin)
+      File.write("/sys/class/gpio/gpio#{pin}/value", value)
+    end
+
+    def pin_set_pud(pin, value)
+      raise NotImplementedError, "Pull up/down not avaliable with this driver. keep it on :off" unless value == :off
+    end
+    
+    def pin_set_trigger(pin, trigger)
+      raise ArgumentError, "trigger should be :falling, :rising, :both or :none" unless [:falling, :rising, :both, :none].include? trigger
+      raise ArgumentError, "Pin #{pin} not exported" unless exported?(pin)
+      File.write("/sys/class/gpio/gpio#{pin}/edge", trigger)
+    end
+
+    def pin_wait_for(pin)
+      fd = File.open("/sys/class/gpio/gpio#{pin}/value", "r")
+      fd.read
+      IO.select(nil, nil, [fd], nil)
+      true
+    end
+
+# Specific behaviours
+
+    def export(pin)
+      File.write("/sys/class/gpio/export", pin)
```

Comment:
I think it should be safe to only allow one object to lock a pin. Having multiple objects on the same pin could result in issues like state problems, deadlocks, etc. I think adding the raise should be good.